src/freenet/clients/fcp/ClientGet.java

```diff
@@ -925,14 +924,17 @@ public boolean filterData() {
 	synchronized RequestStatus getStatus() {
 		boolean totalFinalized = false;
 		int total = 0, min = 0, fetched = 0, fatal = 0, failed = 0;
+		Date latestSuccess = CurrentTimeUTC.get(), latestFailure = null;
```

Comment:
> Why does latestSuccess start at the current time? Is my assumption correct that getStatus() may be called (and indicate progress) without progressPending being set?

This is explained elsewhere:
```
     * Initialized to current time for usability purposes: This allows the user to sort downloads by
     * last success in the user interface to determine which ones are stalling - those will be the
     * ones with the oldest last success date. If we initialized it to "null" only, that would not
     * be possible: The user couldn't distinguish very old stalling downloads from downloads which
     * merely had no success yet because they were added a short time ago */
```

I will add that comment to all the places where it is actually initialized.


> Multiple variables in a single declaration is not allowed by the coding standards.

Will fix. This was done to be coherent with the previous line.